src/clients/c++/request_grpc.cc

```diff
@@ -854,7 +854,8 @@ InferGrpcContextImpl::PreRunProcessing(std::shared_ptr<Request>& request)
       rshared_memory->set_offset(
           reinterpret_cast<InputImpl*>(io.get())->GetSharedMemoryOffset());
       rshared_memory->set_byte_size(
-          reinterpret_cast<InputImpl*>(io.get())->ByteSize());
+          reinterpret_cast<InputImpl*>(io.get())->ByteSize() *
```

Comment:
This should be TotalByteSize... but you've changed that to mean just the total size of the raw bytes. I think you need to fix TotalByteSize to be what it originally was... the total size of an input (i.e. entire batch). See above the call to set_batch_byte_size() is going to be wrong for the shared memory case (and that is needed to correctly support string datatype. You also need a TotalSendByteSize (for http content size and maybe something else but you likely need to make another class member for that).

Reply:
Done